### DISCUSSION TITLE
Middleware cookie verify - working locally but not on Vercel

### DISCUSSION THREAD

**Discussion by Epikur86** [vercel/next.js#40920] [Help]
Hello,

I'm trying to verify a cookie with the library **jose** within the middleware function. When running the code locally it works fine, but as soon as I upload it to Vercel it doesn't. It recognises the cookie, but the verify function throws an error like it is not verified. 
Does someone has an idea how to solve this issue?

```
const secret = process.env.NEXT_PUBLIC_JWT;
export const config = {
  matcher: "/protected/:path*",
};

export async function middleware(req) {
  const cookie = req.cookies.get("key");
  try {
    const ver = await verify(cookie, secret);
    if (req.nextUrl.pathname.startsWith("/protected")) {
      return NextResponse.next();
    } else {
      return NextResponse.redirect(new URL("/notprotected", req.url));
    }
  } catch (e) {
    return NextResponse.redirect(new URL("/", req.url));
  }
}
```

**Comment by icyJoseph** [comment]
What kind of error does it give? Is it a clock tolerance kind of thing? The secret is also available right?

If it is specific to the cookie not being valid for some reason then we need to deep dive into the message and JOSE's docs.

**Comment by Epikur86** [comment]
The secret is available. I put it directly into the function for testing.

Here is the error message:

```
TypeError: The input to be decoded is not correctly encoded.
    at node_modules/jose/dist/browser/runtime/base64url.js:35:0
    at node_modules/jose/dist/browser/jws/flattened/verify.js:78:22
    at node_modules/jose/dist/browser/jws/compact/verify.js:15:27
    at node_modules/jose/dist/browser/jwt/verify.js:6:27
    at pages/cookie.js:39:28
    at middleware.js:20:22
    at node_modules/next/dist/server/web/adapter.js:85:0
    at .:14:47
```
I pasted a valid token (checked via jwt.io) direct into the verify function. On localhost it works and on vercel it doesn't with the same error message like above.

**Reply by Epikur86** [reply]
I found the issue:
before generating the cookie a serialised the token with, I removed and it works now.

```
const serialised = serialize(token, {
    httpOnly: true,
    secure: process.env.NODE_ENV !== "development",
    sameSite: "strict",
    maxAge: 60 * 60 * 24 * 30,
    path: "/",
  });
```
